src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -71,55 +64,979 @@ const ActorInit Bg_Dy_Yoseizo_InitVars = {
+                        break;
+                }
+            }
+        }
+        this->actionFunc = BgDyYoseizo_SetupSpinGrow_Reward;
+        return;
+    }
+
+    globalCtx->envCtx.unk_BF = 2;
+
+    if (globalCtx->sceneNum == SCENE_DAIYOUSEI_IZUMI) {
+        func_800800F8(globalCtx, 0x219B, -99, NULL, 0);
+    } else {
+        func_800800F8(globalCtx, 0x219C, -99, NULL, 0);
+    };
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EV_GREAT_FAIRY_APPEAR);
+    this->actor.draw = BgDyYoseizo_Draw;
+    this->actionFunc = BgDyYoseizo_SetupSpinGrow_NoReward;
+}
+
+// Sets animations for spingrow
+void BgDyYoseizo_SetupSpinGrow_NoReward(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    if (globalCtx->sceneNum == SCENE_DAIYOUSEI_IZUMI) {
+        this->frameCount = Animation_GetLastFrame(&D_06008698);
+        Animation_Change(&this->skelAnime, &D_06008698, 1.0f, 0.0f, this->frameCount, 2, -10.0f);
+    } else {
+        this->frameCount = Animation_GetLastFrame(&D_060031C0);
+        Animation_Change(&this->skelAnime, &D_060031C0, 1.0f, 0.0f, this->frameCount, 2, -10.0f);
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_VO_FR_LAUGH_0);
+    func_8002DF54(globalCtx, &this->actor, 1);
+    this->actionFunc = BgDyYoseizo_SpinGrow_NoReward;
+}
+
+void BgDyYoseizo_SpinGrow_NoReward(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    func_8002DF54(globalCtx, &this->actor, 1);
+    Math_ApproachF(&this->actor.world.pos.y, this->grownHeight, this->heightFraction, 100.0f);
+    Math_ApproachF(&this->scale, 0.035f, this->scaleFraction, 0.005f);
+    Math_ApproachF(&this->heightFraction, 0.8f, 0.1f, 0.02f);
+    Math_ApproachF(&this->scaleFraction, 0.2f, 0.03f, 0.05f);
+    // Finished growing
+    if (this->scale >= 0.034f) {
+        if ((this->actor.shape.rot.y > -8000) && (this->actor.shape.rot.y < 1000)) {
+            SkelAnime_Update(&this->skelAnime);
+            // Turn to front
+            Math_SmoothStepToS(&this->actor.shape.rot.y, 0, 5, 1000, 0);
+            if (fabsf(this->actor.shape.rot.y) < 50.0f) {
+                this->actionFunc = BgDyYoseizo_CompleteSpinGrow_NoReward;
+            }
+        } else {
+            this->actor.shape.rot.y += 3000;
+        }
+    } else {
+        this->actor.shape.rot.y += 3000;
+    }
+    BgDyYoseizo_SpawnParticles(this, globalCtx, 0);
+}
+
+void BgDyYoseizo_CompleteSpinGrow_NoReward(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    func_8002DF54(globalCtx, &this->actor, 1);
+
+    if ((this->frameCount * 1273.0f) <= this->bobTimer) {
+        this->bobTimer = 0.0f;
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if ((this->frameCount <= curFrame) && !(this->animationChanged)) {
+        this->actionFunc = BgDyYoseizo_SetupGreetPlayer_NoReward;
+    }
+}
+
+void BgDyYoseizo_SetupGreetPlayer_NoReward(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    func_8002DF54(globalCtx, &this->actor, (u8)1U);
```

Comment:
hopefully can get rid of cast and U

Reply:
There isn't an enum for csMode yet, right?

Reply:
no there isnt